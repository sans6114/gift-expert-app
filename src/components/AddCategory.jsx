import { useState } from 'react';

import { Button } from 'flowbite-react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setinputValue] = useState('')

    //onCHange
    const onInputChange = ({ target }) => {
        setinputValue(target.value)
    }
    //onSubmit
    const onSubmit = (event) => {
        console.log('submit')
        event.preventDefault()
        //le quito los espacios al valor del input
        const input = inputValue.trim()
        if (input.length <= 1) return
        onNewCategory(input)
        setinputValue('')
    }

    const className = 'bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500'

    return (
        <div className="mb-6">
            <form
                aria-label='form'
                className='flex gap-x-2'
                onSubmit={onSubmit}
            >

                <input
                    onChange={onInputChange}
                    value={inputValue}
                    placeholder="Search category"
                    type="text"
                    id="default-input"
                    className={className}
                />
                <Button type='submit'>Search</Button>
            </form>
        </div>
    )
}


AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
};