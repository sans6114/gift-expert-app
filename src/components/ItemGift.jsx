import { Card } from 'flowbite-react';
import PropTypes from 'prop-types';

export const ItemGift = ({ title, imageUrl }) => {


    return (
        <Card
            className="max-w-sm cursor-pointer"
            imgAlt={`image or gift for image with title: ${title}`}
            imgSrc={imageUrl}
        >
            <h5 className="hidden md:flex text-2xl font-bold tracking-tight text-gray-900 dark:text-white">
                 {title} 
            </h5>
        </Card>
    )
}


ItemGift.propTypes = {
    title: PropTypes.string.isRequired,
    imageUrl: PropTypes.string.isRequired,
};

