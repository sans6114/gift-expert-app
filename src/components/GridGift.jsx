import {
  Button,
  Spinner,
} from 'flowbite-react';
import PropTypes from 'prop-types';

import { useFetchGifts } from '../hooks/useFetchGifts';
import { ItemGift } from './ItemGift';

export const GridGift = ({ category }) => {
    const {images, isLoading} = useFetchGifts(category)

    if (isLoading) {
        return (
        <Button>
            <Spinner aria-label="Spinner button example" size="sm" />
            <span className="pl-3">Loading...</span>
          </Button>
        )
    }
    return (
        <>
            <h3 className='text-3xl'>
                {category}
            </h3>

            <div className="grid grid-cols-2 grid-rows-1 gap-4 md:grid-cols-4 md:grid-rows-1 md:gap-3">

                {
                    images.map(({ id, title, url }) => (
                        <ItemGift key={id} title={title} imageUrl={url} />
                    )
                    )
                }
            </div>


        </>
    )
}

GridGift.propTypes = {
    category: PropTypes.string.isRequired
};

