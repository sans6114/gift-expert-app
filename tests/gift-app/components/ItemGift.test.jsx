import {
  render,
  screen,
} from '@testing-library/react';

import { ItemGift } from '../../../src/components/ItemGift';

describe('test on component <ItemGift />', () => {

    const title = 'Titulo de prueba'
    const imageUrl = 'imagen de prueba'
    const alt = `image or gift for image with title: ${title}`
    test('Debe de hacer match con el snapshot', () => {

        const { container } = render(<ItemGift title={title} imageUrl={imageUrl} />);

        expect(container).toMatchSnapshot()

    });
    test('Debe de mostrar la imagen/gift con el url y alt indicado', () => {

        render(<ItemGift title={title} imageUrl={imageUrl} />);

        const imgElement = screen.getByRole('img');
        // Extraemos solo el pathname de la URL para comparar
        const imgSrcPath = new URL(imgElement.src).pathname;

        // Verificamos 
        expect(imgSrcPath).toBe(`/imagen%20de%20prueba`)
        expect(imgElement.alt).toBe(alt)


    });

    test('debe de mostrar el titulo en el componente', () => {
        render(<ItemGift title={title} imageUrl={imageUrl} />);
        expect(screen.getByText(title)).toBeTruthy()


    });
})