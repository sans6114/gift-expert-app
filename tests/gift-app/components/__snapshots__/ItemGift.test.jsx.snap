// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on component <ItemGift /> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="flex rounded-lg border border-gray-200 bg-white shadow-md dark:border-gray-700 dark:bg-gray-800 flex-col max-w-sm cursor-pointer"
    data-testid="flowbite-card"
  >
    <img
      alt="image or gift for image with title: Titulo de prueba"
      class="rounded-t-lg"
      data-testid="flowbite-card-image"
      src="imagen de prueba"
    />
    <div
      class="flex h-full flex-col justify-center gap-4 p-6"
    >
      <h5
        class="hidden md:flex text-2xl font-bold tracking-tight text-gray-900 dark:text-white"
      >
        Titulo de prueba
      </h5>
    </div>
  </div>
</div>
`;
